### PR TITLE
feat: full support on `PHP8.4`

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Review without `ignore-platform-php`, Reference to #9116 latest checklist.

- [x] fakerphp/php FakerPHP/Faker#904 release on `v1.24`
- [ ] vimeo/psalm vimeo/psalm#10928
- [ ] friendsofphp/php-cs-fixer [PHP-CS-Fixer/PHP-CS-Fixer/milestone/173](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/milestone/173)
- [x] predis/predis predis/predis#1448 release on `v2.3.0`
- [x] kint-php/kint codeigniter4/CodeIgniter4#9289 release on `v6.0`
- [x] mikey179/vfsstream bovigo/vfsStream#290 release on `v1.6.12`

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by ddevsr** [comment]
`FakerPHP/Faker` already merge, release soon on `v1.24`.

**Comment by ddevsr** [comment]
Last dependencies `vimeo/psalm` will be release `v6` in this week

**Comment by paulbalandan** [comment]
How about for php-cs-fixer?

**Comment by ddevsr** [comment]
@paulbalandan Added test again with `PHP_CS_FIXER_IGNORE_ENV`, running smooth

File: .github/workflows/test-coding-standards.yml [hunk 1]
```diff
@@ -58,4 +58,4 @@ jobs:
         run: composer update --ansi --no-interaction
 
       - name: Run lint
-        run: composer cs
+        run: PHP_CS_FIXER_IGNORE_ENV=1 composer cs
```
Comment (paulbalandan) [inline .github/workflows/test-coding-standards.yml:61]:
```suggestion
        run: composer cs
        env:
            PHP_CS_FIXER_IGNORE_ENV: ${{ matrix.php-version == "8.4" }}
```

Comment (paulbalandan) [reply .github/workflows/test-coding-standards.yml]: Sorry, the syntax was not correct. It should be single quotes, not double

Comment (ddevsr) [reply .github/workflows/test-coding-standards.yml]: Rebased again

**Review by michalsn — COMMENTED** [review]
Looks good, thanks for stepping in and taking care of it.

Although I don't think we'll wait for php-cs-fixer to be released - current support is sufficient.

**Review by paulbalandan — APPROVED** [review]

**Review by michalsn — APPROVED** [review]

**Comment by ddevsr** [comment]
@michalsn @paulbalandan Lets merge?

**Comment by paulbalandan** [comment]
Are there no more blocking deps?

**Comment by ddevsr** [comment]
Composer `ignore-platform-req` in utils for `Psalm` only. `PHP-CS-Fixer` just waiting full support, but in this project no issue

**Comment by paulbalandan** [comment]
Ok, our build seems passing so this can be merged.

**Comment by michalsn** [comment]
Thank you @ddevsr!

**Pull request merged**